src/NuGet.Core/NuGet.Protocol/Converters/PackageDependencyGroupConverter.cs

```diff
@@ -32,7 +32,7 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             }
 
             var packages = (set[JsonProperties.Dependencies] as JArray ?? Enumerable.Empty<JToken>())
-                .Select(LoadDependency).ToList();
+                .Select(LoadDependency);
```

Comment:
This is a strategic ToList. 

The Dependencies might get enumerated multiple times, so we avoid the expensive process of loading dependencies which causes a nuspec read.